internal_packages/keybase/lib/pgp-key-store.cjsx

```diff
@@ -0,0 +1,391 @@
+        @trigger(@)
+    )
+
+  getKeybaseData: (key) =>
+    # Given a key, fetches metadata from keybase about that key
+    if not key.key?
+      PGPKeyStore.getKeyContents(key: key)
+    else
+      fingerprint = key.key.get_pgp_fingerprint().toString('hex')
+      kb.getUser(fingerprint, 'key_fingerprint', (user) =>
+        if user?.length == 1
+          key.keybase_user = user[0]
+          @trigger(@)
+      )
+
+  saveNewKey: (address, contents, isPub) =>
+    # Validate the email address, then write to file.
+    if @validAddress(address, isPub)
+      if isPub
+        keyPath = path.join(@_pubKeyDir, address)
+      else
+        keyPath = path.join(@_privKeyDir, address)
+      fs.writeFile(keyPath, contents, (err) =>
+        if (err)
+          @_displayError(err)
+      )
+
+  deleteKey: (key) =>
+    if this._displayDialog(
+        'Delete this key?',
+        'The key will be permanently deleted.',
+        ['Delete', 'Cancel']
+    )
+      fs.unlink(key.path, (err) =>
+        if (err)
+          @_displayError(err)
+      )
+
+  addAddressToKey: (profile, address) =>
+    # NOTE: this is ONLY for public keys
+    if @validAddress(address, true)
+      profile.addresses.push(address)
+      newPath = path.join(@_pubKeyDir, profile.addresses.join(" "))
+      fs.rename(profile.path, newPath, (err) =>
+        if (err)
+          @_displayError(err)
+        )
+
+  removeAddressFromKey: (profile, address) =>
+    # NOTE: this is ONLY for public keys
+    if profile.addresses.length > 1
+      profile.addresses = _.without(profile.addresses, address)
+      newPath = path.join(@_pubKeyDir, profile.addresses.join(" "))
+      fs.rename(profile.path, newPath, (err) =>
+        if (err)
+          @_displayError(err)
+        )
+    else
+      @deleteKey(profile)
+
+  ### Internal Key Management ###
+
+  pubKeys: (address) =>
+    # TODO allow passing list of addresses
+    # fetch public key(s) for an address (synchronous)
+    # if no address, return them all
+    keys = []
+    if not address?
+      keys = @_pubKeys
+    else
+      keys = _.filter @_pubKeys, (key) ->
+        return address in key.addresses
+    keys
+
+  privKeys: ({address, timed}) =>
+    # fetch private key(s) for an address (synchronous).
+    # by default, only return non-timed-out keys
+    # if no address, return them all
+    keys = []
+    if not address?
+      if timed
+        keys = _.filter @_privKeys, (key) ->
+          return key.timeout > Date.now()
+      else
+        keys = @_privKeys
+    else
+      address_keys = _.filter @_privKeys, (key) ->
+        return address in key.addresses
+      if timed
+        keys = _.filter address_keys, (key) ->
+          key.timeout > Date.now()
+      else
+        keys = address_keys
+    return keys
+
+  _displayError: (message) ->
+    dialog = remote.require('dialog')
+    dialog.showErrorBox('Key Management Error', message)
+
+  _displayDialog: (title, message, buttons) ->
```

Comment:
It might be good to remove these in favor of `NylasEnv.showOpenDialog`, which we may have failed to mention :-)